### ISSUE TITLE
Automatically fulfilling digital order lines causes AttributeError in API 

### ISSUE THREAD

**Issue by simon-online** [saleor/saleor#9978]
I'm trying to use automatic fulfillment for digital products that have associated digital content. As they are digital download products with unlimited availability, Track Inventory for these product variants is disabled. It seems that I must have stock assigned in warehouses for this digital product to work otherwise the checkout will return an insufficient stock error.

But when I successfully purchase a digital download product during the checkout process the API raises the following error:
AttributeError: 'NoneType' object has no attribute 'stock' 
at the following line:
https://github.com/saleor/saleor/blob/8e4caaae1d6ffa17fbf0b15985ebb5e2a266f658/saleor/order/actions.py#L604

This stops the automatic fulfillment of the digital order lines. I can workaround this issue by enabling Track Inventory with some stock allocated to a warehouse for these digital product variants and the error doesn't occur which allows the digital products to be automatically fulfilled. However when I do this I also notice that the Fulfillment status for this order doesn't get updated to Fulfilled only the order line is set to Fulfilled even though it is a single item order.

### Steps to reproduce the problem

1. Create a digital product with variant that has Track Inventory disabled but has stock assigned to a warehouse
2. Add a digital content file to that product variant with automatic fulfillment enabled
3. Successfully purchase that digital product variant through the checkout

Here is a Traceback of the error from Saleor v3.3.11:

```
ERROR saleor.graphql.errors.unhandled A query failed unexpectedly [PID:23:Thread-13]
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/graphql/execution/executor.py", line 452, in resolve_or_error
    return executor.execute(resolve_fn, source, info, **args)
  File "/usr/local/lib/python3.9/site-packages/graphql/execution/executors/sync.py", line 16, in execute
    return fn(*args, **kwargs)
  File "/app/saleor/graphql/core/mutations.py", line 374, in mutate
    response = cls.perform_mutation(root, info, **data)
  File "/app/saleor/graphql/checkout/mutations/checkout_complete.py", line 170, in perform_mutation
    order, action_required, action_data = complete_checkout(
  File "/usr/local/lib/python3.9/contextlib.py", line 126, in __exit__
    next(self.gen)
  File "/app/saleor/core/transactions.py", line 18, in transaction_with_commit_on_errors
    error = e
  File "/usr/local/lib/python3.9/contextlib.py", line 126, in __exit__
    next(self.gen)
  File "/app/saleor/core/tracing.py", line 28, in traced_atomic_transaction
    yield
  File "/usr/local/lib/python3.9/site-packages/django/db/transaction.py", line 290, in __exit__
    connection.set_autocommit(True)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 420, in set_autocommit
    self.run_and_clear_commit_hooks()
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 653, in run_and_clear_commit_hooks
    func()
  File "/app/saleor/checkout/complete_checkout.py", line 552, in <lambda>
    lambda: order_created(
  File "/app/saleor/order/actions.py", line 93, in order_created
    order_captured(
  File "/app/saleor/order/actions.py", line 308, in order_captured
    handle_fully_paid_order(manager, order_info, user, app)
  File "/app/saleor/order/actions.py", line 143, in handle_fully_paid_order
    automatically_fulfill_digital_lines(order_info, manager)
  File "/usr/local/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/app/saleor/order/actions.py", line 597, in automatically_fulfill_digital_lines
    line_data.warehouse_pk = allocation.stock.warehouse.pk  # type: ignore
AttributeError: 'NoneType' object has no attribute 'stock'
```

**System information**
Saleor version: v3.3.11

Operating system: Docker Desktop v4.8.2 on MacOS v10.14.6

**Comment by manojLondhe** [comment]
+1

noticed same on Saleor 3.2.x

**Comment by ianninirojas** [comment]
Hi, I'm working with digital products, but the fulfillment is not automatic, there is a special configuration for that? Thanks!

**Comment by simon-online** [comment]
> Hi, I'm working with digital products, but the fulfillment is not automatic, there is a special configuration for that? Thanks!

Unfortunately automatic digital product fulfillment isn't supported in the dashboard. It's only something that can be set up from the API. In general Saleor doesn't have proper support for digital products and there doesn't seem to be anything in the development roadmap to address it either.

If you want good support for digital products and automatic fulfillment, it may be best to look at using another ecommerce solution instead.

**Comment by timuric** [comment]
@simon-online we will add high priority to fix this, hopefully, we can deliver it in 3.5

**Comment by ianninirojas** [comment]
Hi @simon-online did you find a way to do that in the API? Thanks!
